test/integration/hotrestart_test.sh

```diff
@@ -14,34 +14,50 @@ cat test/config/integration/server.json |
 
 # Now start the real server, hot restart it twice, and shut it all down as a basic hot restart
 # sanity test.
+ADMIN_ADDRESS_PATH_0="${TEST_TMPDIR}"/admin_address_file_0.pid
 echo "Starting epoch 0"
 "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" \
-    --restart-epoch 0 --base-id 1 --service-cluster cluster --service-node node &
+    --restart-epoch 0 --base-id 1 --service-cluster cluster --service-node node \
+    -a "${ADMIN_ADDRESS_PATH_0}" &
 
 FIRST_SERVER_PID=$!
 sleep 3
+
+echo "Replacing listener addresses"
+"tools/socket_passing.py" "-o" "${HOT_RESTART_JSON}" "-a" "${ADMIN_ADDRESS_PATH_0}"
+
 # Send SIGUSR1 signal to the first server, this should not kill it. Also send SIGHUP which should
 # get eaten.
 echo "Sending SIGUSR1/SIGHUP to first server"
 kill -SIGUSR1 ${FIRST_SERVER_PID}
 kill -SIGHUP ${FIRST_SERVER_PID}
 sleep 3
 
+ADMIN_ADDRESS_PATH_1="${TEST_TMPDIR}"/admin_address_file_1.pid
```

Comment:
Nit: pid is already a well known term (process id), so maybe use a different suffix, e.g. `.address`.